Do not capture the build-time value of $PATH in the 'emacs' executable
since this can noticeably increase the size of the closure of Emacs
with things like GCC being referenced.

(like emacs-exec-path.patch, but for Emacs 27)

--- a/lisp/loadup.el
+++ b/lisp/loadup.el
@@ -479,7 +479,8 @@ lost after dumping")))
                         ((equal dump-mode "dump") "emacs")
                         ((equal dump-mode "bootstrap") "emacs")
                         ((equal dump-mode "pbootstrap") "bootstrap-emacs.pdmp")
-                        (t (error "unrecognized dump mode %s" dump-mode)))))
+                        (t (error "unrecognized dump mode %s" dump-mode))))
+          (exec-path nil))
       (message "Dumping under the name %s" output)
       (condition-case ()
           (delete-file output)
